sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -315,16 +325,25 @@ def __init__(self, key, secret_uri, label=None, **kwargs):
         self.last_modified = kwargs.get("last_modified", None)
         self.read_only = kwargs.get("read_only", None)
         self.tags = kwargs.get("tags", {})
-        self.value = {"secret_uri": self._secret_uri}
+        self.value = secret_id
+        import six
+        if not self.value:
+            self.value = {}
+        if isinstance(self.value, dict) and "secret_uri" not in self.value.keys():
+            self.value["secret_uri"] = secret_id
+        elif isinstance(self.value, six.string_types):
+            self.value = {"secret_uri": self.value}
```

Comment:
In which case self.value is a string?

Reply:
`SecretReferenceConfigurationSetting("ConnectionString", "myconnstr")` this would have self.value be a string.

Reply:
Then how can user uses self.value, it could be a string or dict?

Reply:
If it's a string, we turn it into a dictionary, same as in FeatFlag.

Reply:
Get it. Seems like we try to behave as smart as we can.

If "secret_uri" is not in the value(dict), we assume value is the value of "secret_uri". 

Is this also the behavior of other languages?

I am concerned it may cause confusions.

Reply:
I believe the other languages return a `ConfigurationSetting` if the value is not the expected format.

Reply:
you mean

new SecretReferenceConfigurationSetting(xxx, yyy) returns a ConfigurationSetting????

Reply:
No, if the user were to do a get on a FeatFlag or SecRet and the value were malformed it would deserialize to a Configurationsetting.

Reply:
Is this the feedback from the latest arch board review?

My understanding is we should give ValueError?